### PR TITLE
Changes from Competition 2023

### PR DESCRIPTION
**Pull request by will-chaba**
### Changelist 

add changes to pedals from competition
- Broadcast brake pedal percentage fixed
- Add raw PAPPS and SAPPS pedal percentage to CAN
- Demote PAPPS/SAPPS disagreement to warning
- OCSC behaviour switched to brake pressure ***MUST BE FIXED WHEN POT ISSUE RESOLVED***
- Display latched faults for IMD, BSPD and AMS (we should really standardize on AMS rather than BMS if that is what we're going to display on the DIM label)
- Min cell voltage displayed on right 7 seg (replacing avg power)

### Testing Done

Existing tests run, test for brake pedal pressed commented out, as hopefully we can resolve the potentiometer issue and revert the OCSC change.
- All existing tests pass
- All changes except current implementation of latched fault broadcast have been tested on the car

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: boards/FSM/Src/Io/Io_Brake.c [hunk 1]
```diff
@@ -3,8 +3,8 @@
 #include "Io_Adc.h"
 
 // TODO set these values
-#define BRAKE_PEDAL_MIN_VOLTAGE (0.5f)
-#define BRAKE_PEDAL_MAX_VOLTAGE (4.5f)
+#define BRAKE_PEDAL_MIN_VOLTAGE (2.08f)
+#define BRAKE_PEDAL_MAX_VOLTAGE (2.310f)
```
Comment (will-chaba) [inline boards/FSM/Src/Io/Io_Brake.c:7]: Note: these numbers will need to be re-tuned when/if we ever fix the potentiometer to stop slipping

File: boards/FSM/Src/Io/Io_Brake.c [hunk 2]
```diff
@@ -73,7 +73,7 @@ float Io_Brake_GetPedalPercentTravel(void)
 {
     float       pedal_voltage    = Io_Brake_GetPedalSensorVoltage();
     const float percent_per_volt = 100 / (BRAKE_PEDAL_MAX_VOLTAGE - BRAKE_PEDAL_MIN_VOLTAGE);
-    return (pedal_voltage - BRAKE_PEDAL_MIN_VOLTAGE) * percent_per_volt;
+    return 100.0f - (pedal_voltage - BRAKE_PEDAL_MIN_VOLTAGE) * percent_per_volt;
```
Comment (will-chaba) [inline boards/FSM/Src/Io/Io_Brake.c:76]: Pedal voltage decreases with pedal travel rather than increase

File: can-bus/json/FSM/FSM_tx.json [hunk 3]
```diff
@@ -43,6 +43,18 @@
 				"max": 100,
 				"unit": "%"
 			},
+			"PappsRawPedalPercentage": {
+				"resolution": 0.1,
+				"min": 0,
+				"max": 100,
+				"unit": "%"
+			},
+			"SappsRawPedalPercentage": {
```
Comment (will-chaba) [inline can-bus/json/FSM/FSM_tx.json:52]: Raw pedal percentage useful for checking OCSC behaviour

File: boards/DIM/Src/App/states/App_DriveState.c [hunk 4]
```diff
@@ -148,7 +139,7 @@ static void DriveStateRunOnTick100Hz(struct StateMachine *const state_machine)
     {
         App_SevenSegDisplays_SetGroupL(seven_seg_displays, speed_kph);
         App_SevenSegDisplays_SetGroupM(seven_seg_displays, instant_power);
-        App_SevenSegDisplays_SetGroupR(seven_seg_displays, avg_power);
+        App_SevenSegDisplays_SetGroupR(seven_seg_displays, min_cell_voltage);
```
Comment (will-chaba) [inline boards/DIM/Src/App/states/App_DriveState.c:142]: @forbesashli should we put avg power on another 7seg? I think speed, instantaneous power and min cell voltage are good for now right?

Comment (forbesashli) [reply boards/DIM/Src/App/states/App_DriveState.c:142]: i think this is also a good choice, due to just needing to see things while driving for testing purposes, i think averaged power is not super useful to most drivers for a while

File: boards/BMS/Src/Io/Io_LatchedFaults.c
Comment (forbesashli) [inline boards/BMS/Src/Io/Io_LatchedFaults.c:1]: was this just to perm latch the fault?

File: boards/FSM/FSM.ioc
Comment (forbesashli) [inline boards/FSM/FSM.ioc:1]: did this change make it on to the car on 05/08/2023?

File: boards/FSM/Src/Io/Io_Brake.c (hunk 1 above)
Comment (forbesashli) [reply boards/FSM/Src/Io/Io_Brake.c:7]: sounds good

**Review by forbesashli — COMMENTED** [review]
made some minor comments just to confirm small things before we push

**Review by Lucien950 — COMMENTED** [review]
Probably should rebase onto the newest commit to work within the refactor

File: boards/FSM/FSM.ioc
Comment (will-chaba) [reply boards/FSM/FSM.ioc:1]: it was a commit from the comp2023 branch that was flashed onto the car, so yes I believe so. I can't remember why pull-up-pull-down was added to the GPIO parameters, I will investigate before merging

File: boards/BMS/Src/Io/Io_LatchedFaults.c
Comment (will-chaba) [reply boards/BMS/Src/Io/Io_LatchedFaults.c:1]: Yeah it latches the fault until a board reset or power cycle

File: firmware/thruna/FSM/Src/App/App_Coolant.c
Comment (Lucien950) [inline firmware/thruna/FSM/Src/App/App_Coolant.c:1]: There is a possible future consideration where this logic is brought closer to the AMS fault

**Review by Lucien950 — APPROVED** [review]
ChopApproveCopApproveGIF

File: firmware/thruna/BMS/Src/Io/Io_LatchedFaults.c
Comment (Lucien950) [inline firmware/thruna/BMS/Src/Io/Io_LatchedFaults.c:1]: image

**Review by Lucien950 — CHANGES_REQUESTED** [review]

File: firmware/thruna/BMS/Src/Io/Io_LatchedFaults.c
Comment (Lucien950) [reply firmware/thruna/BMS/Src/Io/Io_LatchedFaults.c:1]: fixed with cmake reload

**Review by Lucien950 — APPROVED** [review]

**Review by gtaharaedmonds — APPROVED** [review]

File: boards/FSM/FSM.ioc
Comment (will-chaba) [reply boards/FSM/FSM.ioc:1]: Change has no effect on behavior, just a quirk in STM32CUBE, pin was configured as no pull up no pull down both before and after this change. (now it's just explicitly mentioned)

**Pull request merged**